edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -113,6 +114,22 @@ where
             })?;
 
         log::info!("Successfully pulled image {}", image);
+
+        // Now, get the image_id of the image we just pulled for auto-pruning in future
+        match self.list_images().await {
+            Ok(image_name_to_id) => {
+                if image_name_to_id.is_empty() {
+                    log::info!("No docker images present on device");
+                } else {
+                    let image_id = image_name_to_id.get(config.image());
+                    self.migc_persistence
+                        .record_image_use_timestamp(image_id.unwrap_or(&String::default()))
```

Comment:
Do we want to give default string or bubble up error?

Reply:
Seems like we would want to error in this case. @gauravIoTEdge ?

Reply:
The default string was a placeholder till I got around to handling the error. We log a warning and move on:

let image_id = match image_name_to_id.get(config.image()) {
 Some(id) => id,
None => {
log::warn!("bad thing happened");
""
}
};

if !image_id.is_empty() {
self.migc_persistence.record_image_use_timestamp(image_id))
}

Worst case, this image isn't garbage collected (because we have no record of it).